### PR TITLE
Handle realm emojis.

### PR DESCRIPTION
**Pull request by vishwesh3**
fix: #607 

on web
11c3e154-30c3-4677-a1e8-e3ab6e7ede0f

on mobile
11c3e154-30c3-4677-a1e8-e3ab6e7ede0f-2

[message link](https://chat.zulip.org/#narrow/stream/test.20here/subject/realm.20emoji.20in.20Zulip.20Mobile/near/232988)

### CODE REVIEW

**Comment by borisyankov** [comment]
Nice. This is one of the high-value tasks.

File: src/emoji/RealmEmoji.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    return (
+      <Image
+        style={styles.image}
+        source={{ uri: auth.realm.concat(realmEmoji[name].source_url) }}
+      />
+    );
+  }
+}
+
+const mapStateToProps = (state) => ({
```
Comment (borisyankov) [inline src/emoji/RealmEmoji.js]: We stopped using mapStateToProps being specified as a separate function, have a look at the other components in `master`.

File: src/emoji/realmEmojiReducers.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+/* @flow */
+import { StateType, Action } from '../types';
```
Comment (borisyankov) [inline src/emoji/realmEmojiReducers.js]: This is good to be in the 'realm' reducer.

File: src/emoji/realmEmojiReducers.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import {
+  EVENT_REALM_EMOJI_UPDATE,
+  LOGOUT,
+  LOGIN_SUCCESS,
+  ACCOUNT_SWITCH,
+  INIT_REALM_EMOJI,
+} from '../actionConstants';
+
+const initialState = {
+  realm_emoji: {},
```
Comment (borisyankov) [inline src/emoji/realmEmojiReducers.js]: Rename to realmEmoji, we try to keep realm_emoji casing only for API entities. Even better, when in the realm reducer, just rename to `emoji`.

File: src/emoji/RealmEmoji.js (hunk 1 above)
Comment (vishwesh3) [inline src/emoji/RealmEmoji.js]: Done 👍

File: src/emoji/realmEmojiReducers.js (hunk 2 above)
Comment (vishwesh3) [inline src/emoji/realmEmojiReducers.js]: Done 👍

File: src/emoji/realmEmojiReducers.js (hunk 3 above)
Comment (vishwesh3) [inline src/emoji/realmEmojiReducers.js]: Done 👍

**Comment by vishwesh3** [comment]
@borisyankov updated 👍 
Ready to go :runner:

**Comment by borisyankov** [comment]
Nice!

**Pull request merged**